src/main/java/org/broadinstitute/consent/http/resources/DataSetResource.java

```diff
@@ -57,16 +58,28 @@
     private final String END_OF_LINE = System.lineSeparator();
     private final String TSV_DELIMITER = "\t";
     private final DataSetAPI api;
+    private final DatasetService datasetService;
     private final DataAccessRequestAPI dataAccessRequestAPI;
     private final UserService userService;
 
     @Inject
-    public DataSetResource(UserService userService) {
+    public DataSetResource(DatasetService datasetService, UserService userService) {
         this.dataAccessRequestAPI = AbstractDataAccessRequestAPI.getInstance();
         this.api = AbstractDataSetAPI.getInstance();
+        this.datasetService = datasetService;
         this.userService = userService;
     }
 
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/test")
+    @PermitAll
+    public Response createDataset(@Auth AuthUser user, String json) {
+        DataSet dataset = datasetService.createTestDataSet(json);
```

Comment:
In order to pass codacy, I had to actually "use" the service or make it a local variable. I opted to create a mostly empty dataset object (the string here is assigned as its objectId), and return that dataset in the response body (which we are aiming to do by the end with the actual dataset).

Reply:
Nice idea!